packages/model-viewer/src/three-components/ARRenderer.ts

```diff
@@ -293,6 +293,8 @@ export class ARRenderer extends EventDispatcher {
       return;
     }
 
+    this.scene.environment = this[$presentedScene]!.environment;
```

Comment:
It seems [`environment` texture (skybox-image attribute) can be dynamically changed anytime](https://modelviewer.dev/examples/lighting-and-environment.html) so I speculate that copying `.environment` each frame in `$onWebXRFrame()` is safer, rather than doing in `present()` at that `presentedScene` is set. If we do in `present()`, changing environment (skybox-image/) attribute during AR mode won't have an effect. Please let me know if there is any other best place where we should copy `.environment`.

Reply:
I think doing it here is fine. I think for defensive purposes, we should probably check that the environment has changed e.g.,

```typescript
if (this.scene.environment !== this[$presentedScene].environment) {
  this.scene.environment = this[$presentedScene]!.environment;
}
```

That way we don't risk accidentally suggesting to Three.js that the environment is changing in cases where it actually isn't.

Reply:
Thanks for the comment. Updated.

BTW just in case, I release the reference to environment texture at the end of each frame, so the check is seeing whether `presentedScene.environment` is `null` rather than seeing whether the environment has changed since the previous frame.